lib/travis/build/addons/homebrew.rb

```diff
@@ -0,0 +1,97 @@
+require 'travis/build/addons/base'
+require 'shellwords'
+
+module Travis
+  module Build
+    class Addons
+      class Homebrew < Base
+        SUPPORTED_OPERATING_SYSTEMS = %w[
+          osx
+        ].freeze
+
+        def before_prepare?
+          SUPPORTED_OPERATING_SYSTEMS.any? do |os_match|
+            data[:config][:os].to_s == os_match
+          end
+        end
+
+        def before_prepare
+          sh.fold('brew') do
+            update_homebrew if update_homebrew?
+            install_homebrew_packages
+          end
+        end
+
+        private
+
+        def config
+          @config ||= Hash(super)
+        end
+
+        def update_homebrew?
+          config[:update]
+        end
+
+        def update_homebrew
+          sh.echo "Updating Homebrew", ansi: :yellow
+          sh.cmd 'brew update', echo: true, timing: true
+        end
+
+        def config_packages
+          @config_packages ||= Array(config[:packages]).flatten.compact
+        end
+
+        def config_casks
+          @config_casks ||= Array(config[:casks]).flatten.compact
+        end
+
+        def config_taps
+          @config_taps ||= Array(config[:taps]).flatten.compact
+        end
+
+        def create_brewfile?
+          !(config_taps.empty? && config_casks.empty? && config_packages.empty?)
+        end
+
+        def brewfile_contents
```

Comment:
Does it make sense to cache this content for re-use? (There is currently only one place the value is used, so it is not a concern yet.)

Reply:
If we used it again, it would make sense, but I feel like that's not likely to happen. If we needed to use those contents in another way, we would probably be using them through the file.